ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java

```diff
@@ -58,6 +62,16 @@ public static boolean isGzipEncoding(@Nullable final String encodingType) {
       if (fileContentUri.getScheme().startsWith("http")) {
         return getDownloadFileInputStream(context, fileContentUri);
       }
+
+      if (fileContentUriStr.startsWith("data:")) {
+        byte[] decodedDataUrString = Base64.decode(fileContentUriStr.split(",")[1], Base64.DEFAULT);
+        Bitmap bitMap = BitmapFactory.decodeByteArray(decodedDataUrString, 0, decodedDataUrString.length);
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bitMap.compress(Bitmap.CompressFormat.JPEG, 100, bytes);  
```

Comment:
This won't work as it as you're compressing with JPG an image that could be potentially a PNG/GIF or other formats.

Reply:
@cortinico compressFormat also supports webp and PNG! We can add them too!?

Reply:
It seems like the correct approach would be:
```
bitMap.compress(CompressFormat.PNG, 0 /* ignored for PNG */, bytes);
```

Reply:
@cortinico Do you mean something like this?
if(format is PNG)
bitMap.compress(CompressFormat.PNG, 0 /* ignored for PNG */, bytes);
if(format is JPG)
bitMap.compress(CompressFormat.JPG, 0 /* ignored for JPG */, bytes);
if(format is WEBP)
bitMap.compress(CompressFormat.WEBP, 0 /* ignored for WEBP */, bytes);

Reply:
> It seems like the correct approach would be:
> 
> ```
> bitMap.compress(CompressFormat.PNG, 0 /* ignored for PNG */, bytes);
> ```

Or should we do this just for PNG format, Sorry I didn’t completely get your approach

Reply:
The snippet I suggested essentially ignores compression, so you don't need to check the format at all

Reply:
@cortinico I've made the changes. does the changes looks good?